File: mainActivity/src/main/java/com/sierrawireless/avphone/task/ProgressGetUserTask.java
package com.sierrawireless.avphone.task;

import android.app.ProgressDialog;
import android.content.Context;
import android.text.Html;
import android.view.View;
import android.view.Window;
import android.widget.TextView;

import com.sierrawireless.avphone.R;

public class ProgressGetUserTask extends GetUserTask {

    private ProgressDialog dialog;

    ProgressGetUserTask( IUserClient userClient, Context context) {
        super(userClient, context);
    }

    @Override
    protected void onPreExecute() {
        super.onPreExecute();

        int maxProgress = SyncProgress.values().length;
        dialog = new ProgressDialog(getContext());
        dialog.setProgressStyle(ProgressDialog.STYLE_HORIZONTAL);
        dialog.setIndeterminate(false);
        dialog.setMax(maxProgress);
        dialog.setCancelable(false);
        dialog.setTitle(R.string.progress_syncing);

        dialog.setMessage(getContext().getString(R.string.progress_starting));
        dialog.setProgressDrawable(getContext().getResources().getDrawable(
                R.drawable.apptheme_progress_horizontal_holo_light, getContext().getTheme()));

        dialog.setIndeterminateDrawable(getContext().getResources().getDrawable(
                R.drawable.apptheme_progress_indeterminate_horizontal_holo_light, getContext().getTheme()));

        dialog.show();

        // Color has to be set *after* the dialog is shown (see
        // http://blog.supenta.com/2014/07/02/how-to-style-alertdialogs-like-a-pro/)
        int titleDividerId = getContext().getResources().getIdentifier("titleDivider", "id", "android");
        View titleDivider = dialog.findViewById(titleDividerId);
        if (titleDivider != null)
            titleDivider.setBackgroundColor(getContext().getResources().getColor(R.color.sierrared));

        // See http://stackoverflow.com/questions/15271500/how-to-change-alert-dialog-header-divider-color-android
        int alertTitleId = getContext().getResources().getIdentifier("alertTitle", "id", "android");
        Window windows = dialog.getWindow();
        if (windows != null) {
            TextView alertTitle = (TextView) windows.getDecorView().findViewById(alertTitleId);
            alertTitle.setTextColor(getContext().getResources().getColor(R.color.sierrared)); // change title text color
        }
    }

    @Override
    public void onProgressUpdate(GetUserProgress... progress) {
        dialog.setProgress(progress[0].value);
        String stepMessage = getContext().getString(progress[0].stringId);
        String htmlMessage = getContext().getString(R.string.progress_syncing_message, stepMessage);
        dialog.setMessage(Html.fromHtml(htmlMessage));
    }

    @Override
    protected void onPostExecute(GetUserResult result) {
        if (dialog.isShowing()) {
            dialog.dismiss();
        }
        super.onPostExecute(result);
    }

}

File: mainActivity/src/main/java/com/sierrawireless/avphone/task/ProgressGetUserTask.kt

package com.sierrawireless.avphone.task

import android.content.Context

import android.view.View
import android.widget.ProgressBar
import android.widget.TextView
import androidx.appcompat.app.AlertDialog
import androidx.core.content.ContextCompat
import com.sierrawireless.avphone.R
import com.sierrawireless.avphone.activity.MainActivity

class ProgressGetUserTask internal constructor(userClient: IUserClient, context: Context) : GetUserTask(userClient, context) {

    private var dialog: AlertDialog? = null
    private var topProgressBar: ProgressBar? = null
    private var textProgressBar:TextView? = null;

    override fun onPreExecute() {
        super.onPreExecute()

        val maxProgress = SyncProgress.values().size

        val builder: AlertDialog.Builder = AlertDialog.Builder(MainActivity.instance, R.style.Theme_AppCompat_DayNight_Dialog_Alert)
        builder.setCancelable(false) // if you want user to wait for some process to finish,
        builder.setView(com.sierrawireless.avphone.R.layout.layout_loading_dialog)
        dialog = builder.create()


        dialog!!.show()


        // Color has to be set *after* the dialog is shown (see
        // http://blog.supenta.com/2014/07/02/how-to-style-alertdialogs-like-a-pro/)
        val titleDividerId = context.resources.getIdentifier("titleDivider", "id", "android")
        val titleDivider = dialog!!.findViewById<View>(titleDividerId)
        titleDivider?.setBackgroundColor(ContextCompat.getColor(context, R.color.sierrared))

        // See http://stackoverflow.com/questions/15271500/how-to-change-alert-dialog-header-divider-color-android
        val windows = dialog!!.window
        if (windows != null) {
            topProgressBar = windows.decorView.findViewById<ProgressBar> ( R.id.top_progressBar )
            textProgressBar  = windows.decorView.findViewById<TextView> ( R.id.loading_msg )
            topProgressBar!!.isIndeterminate = false
            topProgressBar!!.max = maxProgress
            dialog!!.setCancelable(false)
            dialog!!.setTitle(R.string.progress_syncing)

            textProgressBar!!.text = context.getString(R.string.progress_starting)

            val alertTitle = windows.decorView.findViewById<TextView>(R.id.alert_title)
            alertTitle.text = context.getString(R.string.progress_syncing)
            alertTitle.setTextColor(ContextCompat.getColor(context, R.color.sierrared)) // change title text color
        }
    }


    override fun onPostExecute(result: GetUserResult) {
        if (!MainActivity.instance.isFinishing && !MainActivity.instance.isDestroyed) {
            try {
                if (dialog != null && dialog!!.isShowing) {
                    dialog!!.dismiss()
                }
            } catch (e: IllegalArgumentException) {
            } catch (e: Exception) {
            } finally {
                dialog = null
            }
        }
        super.onPostExecute(result)
    }

}
